src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1896,9 +1896,17 @@ protected override void InvokeDefaultAction(string path)
                             }
 
                             bool hidden = false;
+                            bool checkReparsePoint = true;
                             if (!Force)
                             {
                                 hidden = (recursiveDirectory.Attributes & FileAttributes.Hidden) != 0;
+
+                                // Performance optimization.
+                                // Since we have already checked Attributes for Hidden we have already did a p/invoke
+                                // and initialized Attributes property.
+                                // So here we can check for ReparsePoint without new p/invoke.
+                                // If it is not a reparse point we skip one p/invoke in IsReparsePointLikeSymlink() below.
+                                checkReparsePoint = (recursiveDirectory.Attributes & FileAttributes.ReparsePoint) != 0;
```

Comment:
Isn't `HasFlag()` now performance optimised, and so preferable?

Reply:
I keep the file style.